src/System.Threading.Tasks.Extensions/pkg/System.Threading.Tasks.Extensions.pkgproj

```diff
@@ -19,4 +19,15 @@
     <SuppressMetaPackage Include="NETStandard.Library" />
   </ItemGroup>
   <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.targets))\dir.targets" />
+  <ItemGroup>
+    <!--
+      All frameworks except net45 support the frozen 4.2.0.0 api version. Need to include this after dir.targets to
+      have DefaultValidateionFramework defined. Normally this wouldn't be necessary but the harvested supported
```

Comment:
Spelling.  Also I don't like using DefaultValidateFramework, can you instead you the string from above and place this next to the project reference? Also you might want to reference an issue to clean it up.
```
<SupportedFramework Include="netcore45;netcoreapp1.0;wpa81;wp8;$(AllXamarinFrameworks)">
  <Version>4.2.0.0</Version>
</SupportedFramework>
```

Reply:
`netcore45;netcoreapp1.0;wpa81;wp8;$(AllXamarinFrameworks)` for some reason when directly using SupportedFramework it doesn't expand all the versions which is why I switched to using DefaultValidateFramework item group instead.

Reply:
Tell me more about expand all the versions

Reply:
When I copied the set of frameworks down to here I got all these errors. I tried to iterate and add them but I quickly ended up with almost all of DefaultValidateFramework so I just switched to that instead of listing every version of every target. 

```
   2:2>D:\git\corefx\bin\testPkg\tools\sdk\2.1.300-preview2-008530\Sdks\Microsoft.NET.Sdk\targets\Microsoft.PackageDependencyResolution.targets(205,5): error MSB4018: The "ResolvePackageAssets" task failed unexpectedly. [D:\git\corefx\bin\testPkg\projects\System.Composition.Hosting\netstandard1.4\project.csproj] [D:\git\corefx\pkg\test\testPackages.proj]
   2:2>D:\git\corefx\bin\testPkg\tools\sdk\2.1.300-preview2-008530\Sdks\Microsoft.NET.Sdk\targets\Microsoft.PackageDependencyResolution.targets(205,5): error MSB4018: System.IO.IOException: The process cannot access the file 'D:\git\corefx\bin\testPkg\projects\System.Composition.Hosting\netstandard1.4\obj\Debug\netstandard1.4\project.assets.cache' because it is being used by another process. [D:\git\corefx\bin\testPkg\projects\System.Composition.Hosting\netstandard1.4\project.csproj] [D:\git\corefx\pkg\test\testPackages.proj]
   2:2>D:\git\corefx\bin\testPkg\tools\sdk\2.1.300-preview2-008530\Sdks\Microsoft.NET.Sdk\targets\Microsoft.PackageDependencyResolution.targets(205,5): error MSB4018: at System.IO.FileStream.ValidateFileHandle(SafeFileHandle fileHandle) [D:\git\corefx\bin\testPkg\projects\System.Composition.Hosting\netstandard1.4\project.csproj] [D:\git\corefx\pkg\test\testPackages.proj]
   2:2>D:\git\corefx\bin\testPkg\tools\sdk\2.1.300-preview2-008530\Sdks\Microsoft.NET.Sdk\targets\Microsoft.PackageDependencyResolution.targets(205,5): error MSB4018: at System.IO.FileStream.CreateFileOpenHandle(FileMode mode, FileShare share, FileOptions options) [D:\git\corefx\bin\testPkg\projects\System.Composition.Hosting\netstandard1.4\project.csproj] [D:\git\corefx\pkg\test\testPackages.proj]
   2:2>D:\git\corefx\bin\testPkg\tools\sdk\2.1.300-preview2-008530\Sdks\Microsoft.NET.Sdk\targets\Microsoft.PackageDependencyResolution.targets(205,5): error MSB4018: at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options) [D:\git\corefx\bin\testPkg\projects\System.Composition.Hosting\netstandard1.4\project.csproj] [D:\git\corefx\pkg\test\testPackages.proj]
   2:2>D:\git\corefx\bin\testPkg\tools\sdk\2.1.300-preview2-008530\Sdks\Microsoft.NET.Sdk\targets\Microsoft.PackageDependencyResolution.targets(205,5): error MSB4018: at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share) [D:\git\corefx\bin\testPkg\projects\System.Composition.Hosting\netstandard1.4\project.csproj] [D:\git\corefx\pkg\test\testPackages.proj]
   2:2>D:\git\corefx\bin\testPkg\tools\sdk\2.1.300-preview2-008530\Sdks\Microsoft.NET.Sdk\targets\Microsoft.PackageDependencyResolution.targets(205,5): error MSB4018: at System.IO.File.Open(String path, FileMode mode, FileAccess access, FileShare share) [D:\git\corefx\bin\testPkg\projects\System.Composition.Hosting\netstandard1.4\project.csproj] [D:\git\corefx\pkg\test\testPackages.proj]
   2:2>D:\git\corefx\bin\testPkg\tools\sdk\2.1.300-preview2-008530\Sdks\Microsoft.NET.Sdk\targets\Microsoft.PackageDependencyResolution.targets(205,5): error MSB4018: at Microsoft.NET.Build.Tasks.ResolvePackageAssets.CacheWriter..ctor(ResolvePackageAssets task) [D:\git\corefx\bin\testPkg\projects\System.Composition.Hosting\netstandard1.4\project.csproj] [D:\git\corefx\pkg\test\testPackages.proj]
   2:2>D:\git\corefx\bin\testPkg\tools\sdk\2.1.300-preview2-008530\Sdks\Microsoft.NET.Sdk\targets\Microsoft.PackageDependencyResolution.targets(205,5): error MSB4018: at Microsoft.NET.Build.Tasks.ResolvePackageAssets.CacheReader..ctor(ResolvePackageAssets task) [D:\git\corefx\bin\testPkg\projects\System.Composition.Hosting\netstandard1.4\project.csproj] [D:\git\corefx\pkg\test\testPackages.proj]
   2:2>D:\git\corefx\bin\testPkg\tools\sdk\2.1.300-preview2-008530\Sdks\Microsoft.NET.Sdk\targets\Microsoft.PackageDependencyResolution.targets(205,5): error MSB4018: at Microsoft.NET.Build.Tasks.ResolvePackageAssets.ReadItemGroups() [D:\git\corefx\bin\testPkg\projects\System.Composition.Hosting\netstandard1.4\project.csproj] [D:\git\corefx\pkg\test\testPackages.proj]
   2:2>D:\git\corefx\bin\testPkg\tools\sdk\2.1.300-preview2-008530\Sdks\Microsoft.NET.Sdk\targets\Microsoft.PackageDependencyResolution.targets(205,5): error MSB4018: at Microsoft.NET.Build.Tasks.ResolvePackageAssets.ExecuteCore() [D:\git\corefx\bin\testPkg\projects\System.Composition.Hosting\netstandard1.4\project.csproj] [D:\git\corefx\pkg\test\testPackages.proj]
   2:2>D:\git\corefx\bin\testPkg\tools\sdk\2.1.300-preview2-008530\Sdks\Microsoft.NET.Sdk\targets\Microsoft.PackageDependencyResolution.targets(205,5): error MSB4018: at Microsoft.NET.Build.Tasks.TaskBase.Execute() [D:\git\corefx\bin\testPkg\projects\System.Composition.Hosting\netstandard1.4\project.csproj] [D:\git\corefx\pkg\test\testPackages.proj]
   2:2>D:\git\corefx\bin\testPkg\tools\sdk\2.1.300-preview2-008530\Sdks\Microsoft.NET.Sdk\targets\Microsoft.PackageDependencyResolution.targets(205,5): error MSB4018: at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() [D:\git\corefx\bin\testPkg\projects\System.Composition.Hosting\netstandard1.4\project.csproj] [D:\git\corefx\pkg\test\testPackages.proj]
   2:2>D:\git\corefx\bin\testPkg\tools\sdk\2.1.300-preview2-008530\Sdks\Microsoft.NET.Sdk\targets\Microsoft.PackageDependencyResolution.targets(205,5): error MSB4018: at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(ITaskExecutionHost taskExecutionHost, TaskLoggingContext taskLoggingContext, TaskHost taskHost, ItemBucket bucket, TaskExecutionMode howToExecuteTask) [D:\git\corefx\bin\testPkg\projects\System.Composition.Hosting\netstandard1.4\project.csproj] [D:\git\corefx\pkg\test\testPackages.proj]
   2:2>D:\git\corefx\pkg\test\testPackages.proj(177,5): error MSB3073: The command "D:\git\corefx\bin/testPkg/tools//dotnet msbuild /t:Test "D:\git\corefx\bin/testPkg/test.msbuild" /p:TestPackages=System.Composition.Hosting" exited with code 1.
   2:2>D:\git\corefx\bin\testPkg\packageTest.targets(104,5): error MSB3371: The file "obj\Debug\netstandard1.4\\.testComplete" cannot be created. The process cannot access the file 'D:\git\corefx\bin\testPkg\projects\System.Composition.TypedParts\netstandard1.4\obj\Debug\netstandard1.4\.testComplete' because it is being used by another process. [D:\git\corefx\bin\testPkg\projects\System.Composition.TypedParts\netstandard1.4\project.csproj] [D:\git\corefx\pkg\test\testPackages.proj]
   2:2>D:\git\corefx\pkg\test\testPackages.proj(177,5): error MSB3073: The command "D:\git\corefx\bin/testPkg/tools//dotnet msbuild /t:Test "D:\git\corefx\bin/testPkg/test.msbuild" /p:TestPackages=System.Composition.TypedParts" exited with code 1.
   2:2>D:\git\corefx\bin\testPkg\packageTest.targets(69,5): error : Assembly 'System.Data.SqlClient' has insufficient version for dependency 'System.Diagnostics.DiagnosticSource' : 4.0.3.0 < 4.0.4.0. [D:\git\corefx\bin\testPkg\projects\System.Data.SqlClient\netcoreapp2.2\project.csproj] [D:\git\corefx\pkg\test\testPackages.proj]
   2:2>D:\git\corefx\bin\testPkg\packageTest.targets(69,5): error : Assembly 'System.Data.SqlClient' has insufficient version for dependency 'System.Diagnostics.DiagnosticSource' : 4.0.3.0 < 4.0.4.0. [D:\git\corefx\bin\testPkg\projects\System.Data.SqlClient\netcoreapp2.2\project.csproj] [D:\git\corefx\pkg\test\testPackages.proj]
   2:2>D:\git\corefx\bin\testPkg\packageTest.targets(69,5): error : Assembly 'System.Data.SqlClient' has insufficient version for dependency 'System.Diagnostics.DiagnosticSource' : 4.0.3.0 < 4.0.4.0. [D:\git\corefx\bin\testPkg\projects\System.Data.SqlClient\netcoreapp2.2\project.csproj] [D:\git\corefx\pkg\test\testPackages.proj]
   2:2>D:\git\corefx\bin\testPkg\packageTest.targets(69,5): error : Assembly 'System.Data.SqlClient' has insufficient version for dependency 'System.Diagnostics.DiagnosticSource' : 4.0.3.0 < 4.0.4.0. [D:\git\corefx\bin\testPkg\projects\System.Data.SqlClient\netcoreapp2.2\project.csproj] [D:\git\corefx\pkg\test\testPackages.proj]
   2:2>D:\git\corefx\bin\testPkg\packageTest.targets(69,5): error : Assembly 'System.Data.SqlClient' has insufficient version for dependency 'System.Diagnostics.DiagnosticSource' : 4.0.3.0 < 4.0.4.0. [D:\git\corefx\bin\testPkg\projects\System.Data.SqlClient\netcoreapp2.2\project.csproj] [D:\git\corefx\pkg\test\testPackages.proj]
   2:2>D:\git\corefx\bin\testPkg\packageTest.targets(69,5): error : Assembly 'System.Data.SqlClient' has insufficient version for dependency 'System.Diagnostics.DiagnosticSource' : 4.0.3.0 < 4.0.4.0. [D:\git\corefx\bin\testPkg\projects\System.Data.SqlClient\netcoreapp2.2\project.csproj] [D:\git\corefx\pkg\test\testPackages.proj]
   2:2>D:\git\corefx\bin\testPkg\packageTest.targets(69,5): error : Assembly 'System.Data.SqlClient' has insufficient version for dependency 'System.Diagnostics.DiagnosticSource' : 4.0.3.0 < 4.0.4.0. [D:\git\corefx\bin\testPkg\projects\System.Data.SqlClient\netcoreapp2.2\project.csproj] [D:\git\corefx\pkg\test\testPackages.proj]
   2:2>D:\git\corefx\bin\testPkg\packageTest.targets(69,5): error : Assembly 'System.Data.SqlClient' has insufficient version for dependency 'System.Diagnostics.DiagnosticSource' : 4.0.3.0 < 4.0.4.0. [D:\git\corefx\bin\testPkg\projects\System.Data.SqlClient\netcoreapp2.2\project.csproj] [D:\git\corefx\pkg\test\testPackages.proj]
   2:2>D:\git\corefx\bin\testPkg\packageTest.targets(69,5): error : Assembly 'System.Data.SqlClient' has insufficient version for dependency 'System.Diagnostics.DiagnosticSource' : 4.0.3.0 < 4.0.4.0. [D:\git\corefx\bin\testPkg\projects\System.Data.SqlClient\netcoreapp2.2\project.csproj] [D:\git\corefx\pkg\test\testPackages.proj]
   2:2>D:\git\corefx\bin\testPkg\packageTest.targets(69,5): error : Assembly 'System.Data.SqlClient' has insufficient version for dependency 'System.Diagnostics.DiagnosticSource' : 4.0.3.0 < 4.0.4.0. [D:\git\corefx\bin\testPkg\projects\System.Data.SqlClient\netcoreapp2.2\project.csproj] [D:\git\corefx\pkg\test\testPackages.proj]
   2:2>D:\git\corefx\bin\testPkg\packageTest.targets(69,5): error : Assembly 'System.Data.SqlClient' has insufficient version for dependency 'System.Diagnostics.DiagnosticSource' : 4.0.3.0 < 4.0.4.0. [D:\git\corefx\bin\testPkg\projects\System.Data.SqlClient\netcoreapp2.2\project.csproj] [D:\git\corefx\pkg\test\testPackages.proj]
   2:2>D:\git\corefx\bin\testPkg\packageTest.targets(69,5): error : Assembly 'System.Data.SqlClient' has insufficient version for dependency 'System.Diagnostics.DiagnosticSource' : 4.0.3.0 < 4.0.4.0. [D:\git\corefx\bin\testPkg\projects\System.Data.SqlClient\netcoreapp2.2\project.csproj] [D:\git\corefx\pkg\test\testPackages.proj]
   2:2>D:\git\corefx\pkg\test\testPackages.proj(177,5): error MSB3073: The command "D:\git\corefx\bin/testPkg/tools//dotnet msbuild /t:Test "D:\git\corefx\bin/testPkg/test.msbuild" /p:TestPackages=System.Data.SqlClient" exited with code 1.
   2:2>D:\git\corefx\bin\testPkg\packageTest.targets(37,5): error : Assembly 'System.Reflection.Metadata' has insufficient version for dependency 'System.Collections.Immutable' : 1.2.3.0 < 1.2.4.0. [D:\git\corefx\bin\testPkg\projects\System.Reflection.Metadata\netcoreapp2.2\project.csproj] [D:\git\corefx\pkg\test\testPackages.proj]
   2:2>D:\git\corefx\bin\testPkg\packageTest.targets(69,5): error : Assembly 'System.Reflection.Metadata' has insufficient version for dependency 'System.Collections.Immutable' : 1.2.3.0 < 1.2.4.0. [D:\git\corefx\bin\testPkg\projects\System.Reflection.Metadata\netcoreapp2.2\project.csproj] [D:\git\corefx\pkg\test\testPackages.proj]
   2:2>D:\git\corefx\bin\testPkg\packageTest.targets(69,5): error : Assembly 'System.Reflection.Metadata' has insufficient version for dependency 'System.Collections.Immutable' : 1.2.3.0 < 1.2.4.0. [D:\git\corefx\bin\testPkg\projects\System.Reflection.Metadata\netcoreapp2.2\project.csproj] [D:\git\corefx\pkg\test\testPackages.proj]
   2:2>D:\git\corefx\bin\testPkg\packageTest.targets(69,5): error : Assembly 'System.Reflection.Metadata' has insufficient version for dependency 'System.Collections.Immutable' : 1.2.3.0 < 1.2.4.0. [D:\git\corefx\bin\testPkg\projects\System.Reflection.Metadata\netcoreapp2.2\project.csproj] [D:\git\corefx\pkg\test\testPackages.proj]
   2:2>D:\git\corefx\bin\testPkg\packageTest.targets(69,5): error : Assembly 'System.Reflection.Metadata' has insufficient version for dependency 'System.Collections.Immutable' : 1.2.3.0 < 1.2.4.0. [D:\git\corefx\bin\testPkg\projects\System.Reflection.Metadata\netcoreapp2.2\project.csproj] [D:\git\corefx\pkg\test\testPackages.proj]
   2:2>D:\git\corefx\bin\testPkg\packageTest.targets(69,5): error : Assembly 'System.Reflection.Metadata' has insufficient version for dependency 'System.Collections.Immutable' : 1.2.3.0 < 1.2.4.0. [D:\git\corefx\bin\testPkg\projects\System.Reflection.Metadata\netcoreapp2.2\project.csproj] [D:\git\corefx\pkg\test\testPackages.proj]
   2:2>D:\git\corefx\bin\testPkg\packageTest.targets(69,5): error : Assembly 'System.Reflection.Metadata' has insufficient version for dependency 'System.Collections.Immutable' : 1.2.3.0 < 1.2.4.0. [D:\git\corefx\bin\testPkg\projects\System.Reflection.Metadata\netcoreapp2.2\project.csproj] [D:\git\corefx\pkg\test\testPackages.proj]
   2:2>D:\git\corefx\bin\testPkg\packageTest.targets(69,5): error : Assembly 'System.Reflection.Metadata' has insufficient version for dependency 'System.Collections.Immutable' : 1.2.3.0 < 1.2.4.0. [D:\git\corefx\bin\testPkg\projects\System.Reflection.Metadata\netcoreapp2.2\project.csproj] [D:\git\corefx\pkg\test\testPackages.proj]
   2:2>D:\git\corefx\bin\testPkg\packageTest.targets(69,5): error : Assembly 'System.Reflection.Metadata' has insufficient version for dependency 'System.Collections.Immutable' : 1.2.3.0 < 1.2.4.0. [D:\git\corefx\bin\testPkg\projects\System.Reflection.Metadata\netcoreapp2.2\project.csproj] [D:\git\corefx\pkg\test\testPackages.proj]
   2:2>D:\git\corefx\bin\testPkg\packageTest.targets(69,5): error : Assembly 'System.Reflection.Metadata' has insufficient version for dependency 'System.Collections.Immutable' : 1.2.3.0 < 1.2.4.0. [D:\git\corefx\bin\testPkg\projects\System.Reflection.Metadata\netcoreapp2.2\project.csproj] [D:\git\corefx\pkg\test\testPackages.proj]
   2:2>D:\git\corefx\bin\testPkg\packageTest.targets(69,5): error : Assembly 'System.Reflection.Metadata' has insufficient version for dependency 'System.Collections.Immutable' : 1.2.3.0 < 1.2.4.0. [D:\git\corefx\bin\testPkg\projects\System.Reflection.Metadata\netcoreapp2.2\project.csproj] [D:\git\corefx\pkg\test\testPackages.proj]
   2:2>D:\git\corefx\bin\testPkg\packageTest.targets(69,5): error : Assembly 'System.Reflection.Metadata' has insufficient version for dependency 'System.Collections.Immutable' : 1.2.3.0 < 1.2.4.0. [D:\git\corefx\bin\testPkg\projects\System.Reflection.Metadata\netcoreapp2.2\project.csproj] [D:\git\corefx\pkg\test\testPackages.proj]
   2:2>D:\git\corefx\bin\testPkg\packageTest.targets(69,5): error : Assembly 'System.Reflection.Metadata' has insufficient version for dependency 'System.Collections.Immutable' : 1.2.3.0 < 1.2.4.0. [D:\git\corefx\bin\testPkg\projects\System.Reflection.Metadata\netcoreapp2.2\project.csproj] [D:\git\corefx\pkg\test\testPackages.proj]
   2:2>D:\git\corefx\pkg\test\testPackages.proj(177,5): error MSB3073: The command "D:\git\corefx\bin/testPkg/tools//dotnet msbuild /t:Test "D:\git\corefx\bin/testPkg/test.msbuild" /p:TestPackages=System.Reflection.Metadata" exited with code 1.
   2:2>D:\git\corefx\bin\testPkg\packageTest.targets(104,5): error MSB3371: The file "obj\Debug\\.testComplete" cannot be created. The process cannot access the file 'D:\git\corefx\bin\testPkg\projects\System.Threading.AccessControl\xamarintvos10\obj\Debug\.testComplete' because it is being used by another process. [D:\git\corefx\bin\testPkg\projects\System.Threading.AccessControl\xamarintvos10\project.csproj] [D:\git\corefx\pkg\test\testPackages.proj]
   2:2>D:\git\corefx\pkg\test\testPackages.proj(177,5): error MSB3073: The command "D:\git\corefx\bin/testPkg/tools//dotnet msbuild /t:Test "D:\git\corefx\bin/testPkg/test.msbuild" /p:TestPackages=System.Threading.AccessControl" exited with code 1.
```

Reply:
Filed issue https://github.com/dotnet/corefx/issues/29383 to remove this once we update the harvested versions. Also filed https://github.com/dotnet/buildtools/issues/2025 to look into why we aren't correctly prompting the versions when we harvest supported versions.